modules/ingester/trace.go

```diff
@@ -42,5 +51,18 @@ func (t *trace) Push(_ context.Context, trace []byte) error {
 
 	t.traceBytes.Traces = append(t.traceBytes.Traces, trace)
 
+	if searchData != nil {
+		searchDataSize := len(searchData)
+		if t.maxSearchBytes != 0 { // disable limit if set to 0
+			if t.currentSearchBytes+searchDataSize > t.maxSearchBytes {
+				// todo: info level since we are not expecting this limit to be hit, but calibrate accordingly in the future
+				level.Info(cortex_util.Logger).Log("msg", "size of search data exceeded max search bytes limit", "maxSearchBytes", t.maxSearchBytes)
```

Comment:
we will need to metric this so we can track it ourselves as well as eventually expose it to people in Grafana Cloud